File: aulas/desafioEmails/email.py
class Email:

    def __init__(self):
        self.aluno = ""
        self.email = ""

    def gerar_email(self, nome):
        partes = nome.lower().split(" ")

        if len(partes) > 1:
            return f"{partes[0]}.{partes[-1]}@ufn.edu.br"

        return f"{partes[0]}@ufn.edu.br"

    def set_aluno(self, nome):
        self.aluno = nome
        self.email = self.gerar_email(nome)

    def get_aluno(self):
        return self.aluno

    def get_email(self):
        return self.email

File: aulas/desafioEmails/Email.java
public class Email{
   private String aluno;
   private String email;

   public void aluno(String nome){
    this.aluno = nome;
    this.email = gerarEmail(nome);
   }


    private String gerarEmail(String email){
        String [] partes = aluno.toLowerCase().split(" ");
        if(partes.length > 1){
        return partes[0] + "." + partes[partes.length - 1] + "@ufn.edu.br";
        }
        return partes[0] + "@ufn.edu.br";
    }

    public String getAluno(){return aluno;}
    public String getEmail(){return email;}

    public void setAluno(String aluno){
        this.aluno = aluno;
        this.email = gerarEmail(aluno);
    }
    public void setEmail(String email){this.email = email;}

    @Override
    public String toString(){
        return "Aluno: " + aluno + "email: " + email;
    }
}